deploymentTools/deploy-all-stacks.sh

```diff
@@ -0,0 +1,123 @@
+    
+    deploy_command="sam deploy --config-env dev --resolve-s3 --stack-name $DevStackName --parameter-overrides \"CodeSigningConfigArn=none Environment=dev PermissionsBoundary=none SecretPrefix=none VpcStackName=vpc-cri L2DynamoStackName=$L2DynamoStackName L2KMSStackName=$L2KMSStackName\" --capabilities CAPABILITY_IAM"
+    run_deployment_in_parallel "$DevStackName" "../deploy" "$deploy_command" &
+fi
+
+# Wait for Layer 3 deployments to finish
+wait
+
+# Loop through the selected stack names and create deployment commands for the rest
+for stack_name in "${stacksToDeploy[@]}"; do
+    case "$stack_name" in
+        "$L2KMSStackName" | "$L2DynamoStackName" | "$IPVStackName" | "$DevStackName")
+            # Skip L2KMSStackName, L2DynamoStackName, IPVStackName, and DevStackName as they were deployed earlier
+            continue
+            ;;
+        "$TestHarnessStackName" | "$YotiStubStackName" | "$GovNotifyStackName")
+            # Pass the 'sam deploy' command for TestHarnessStackName, YotiStubStackName, and GovNotifyStackName
+            
+            # Pass the correct directory to run_deployment_in_parallel based on the stack name
+            case "$stack_name" in
+                "$TestHarnessStackName")
+                    dir="../test-harness"
+                    deploy_command="sam deploy --config-env dev --resolve-s3 --stack-name $TestHarnessStackName --parameter-overrides \"CodeSigningConfigArn=none Environment=dev PermissionsBoundary=none SecretPrefix=none VpcStackName=vpc-cri BackendStack=$DevStackName\" --capabilities CAPABILITY_IAM CAPABILITY_NAMED_IAM"
+                    ;;
+                "$YotiStubStackName")
+                    dir="../yoti-stub"
+                    deploy_command="sam deploy --config-env dev --resolve-s3 --stack-name $YotiStubStackName --parameter-overrides \"CodeSigningConfigArn=none Environment=dev PermissionsBoundary=none SecretPrefix=none VpcStackName=vpc-cri L2DynamoStackName=$L2DynamoStackName L2KMSStackName=$L2KMSStackName\" --capabilities CAPABILITY_IAM"
+                    ;;
+                "$GovNotifyStackName")
+                    dir="../gov-notify-stub"
+                    deploy_command="sam deploy --config-env dev --resolve-s3 --stack-name $GovNotifyStackName --parameter-overrides \"CodeSigningConfigArn=none Environment=dev PermissionsBoundary=none SecretPrefix=none VpcStackName=vpc-cri L2DynamoStackName=$L2DynamoStackName L2KMSStackName=$L2KMSStackName\" --capabilities CAPABILITY_IAM"
+                    ;;
+            esac
+            
+            # Create deployment commands for TestHarnessStackName, YotiStubStackName, and GovNotifyStackName and run them in parallel
+            run_deployment_in_parallel "$stack_name" "$dir" "$deploy_command" &
+            ;;
+    esac
+done
+
+# Wait for all deployments to finish
+wait
+
+echo ""
+echo "🎉 Following Stacks Deployed Successfully: ${stacksToDeploy[*]}!
```

Comment:
Just wondering if anything fails what happens? Eg what if the dynamo deploy fails which will cause the main stack deploy to fail and then the test harness deploy to fail

Reply:
The script isn't clever enough yet to account for failures it assumes all the stacks deploy successfully and if not then the failed stack deployed will end and the next one will deploy (which if it depends on the one before it will also fail obviously)

Reply:
I'll look to make it cleverer to listen out for stack deployment failures and terminate the script execution at any time a deployment fails for whatever reason

Reply:
done :)